### ISSUE TITLE
Improvement - Layouts: Dynamic Preview

### ISSUE THREAD

**Issue by ermincelikovic** [acf-extended/ACF-Extended#29]
**Is your feature request related to a problem? Please describe.**
At the moment, the dynamic preview of the layouts tends not to be too accurate since the styles and the scripts of the theme itself are not included in the preview. This is mostly reflected on the fonts in my case - but I can imagine that this could also be an issue for the usage of libraries like bootstrap.

**Describe the solution you'd like**
The cleanest solution for the improvement (as far as I see it) is adding `wp_head()` and `wp_footer()` to the layout structure and wrapping it all in iframe to prevent the overwriting of the wp-admin styles.

**Describe alternatives you've considered**
I've tried to make a function which would do exactly this - but using javascript to append an iframe. Works quite fine when the modules are not having any javascript internally - but in case of javascript, the render is unreliable.

**Additional context**
Not urgent, but definitely nice to have. If I can assist anyhow, let me know.

**Comment by acf-extended** [comment]
Hello,

Thanks for the feedback!

The problem is quite similar to what Gutenberg developers have to deal with. The thing is that Dynamic Preview/Gutenberg force you to split your template between "sections" which are independant from each other. It is kind of an another way to build your template.

While your solution could work in some edge cases, it is kinda "hacky", as it could lead to some nasty performance problems. For example, if you got 10 layouts on the admin page, the browser will load 10 times all header + Footer JS/CSS, as each iframe are processed as an individual page by the browser.

I would recommend to manually enqueue the library you use in the admin screen, and only when necessary. If you're looking for filters & actions, you can enqueue specific files only once per flexible content using:

```php
add_action('acfe/flexible/enqueue/name=my_flexible', 'my_flexible_enqueue', 10, 2);
function my_flexible_enqueue($field, $is_preview){

    // wp_enqueue_style();
    // wp_enqueue_script();

}
```

Or, you can enqueue files with specific layouts only:

```php
add_action('acfe/flexible/layout/enqueue/name=my_flexible&layout=my-layout', 'my_flexible_layout_enqueue', 10, 3);
function my_flexible_layout_enqueue($field, $layout, $is_preview){

    // wp_enqueue_style()
    // wp_enqueue_script()

}
```

Have a nice day!

Regards.

**Comment by ermincelikovic** [comment]
Thank you for taking your time for the answer!

Doing the script/style enqueue is also working well only in the edge cases - it's mostly overwriting the wp-admin. At the same time, if the theme styles are not included, the "preview" tends not to be correct.

Maybe something in between would work? An option for selective external styles + iframe wrapper?

**Comment by acf-extended** [comment]
Hello,

The iframe is also problematic when it comes to live preview. Example: make a change in a field input and display the change in live, without reloading. I really don't think this is a suitable solution.

Regards.
